File: project7/python/tasks.py
import json
import re
from fragAndReplicate import fragment_and_replicate
from pyspark import SparkContext

def task1(playRDD):
    return playRDD

def task2_flatmap(x):
    return []

def task3(nobelRDD):
    return nobelRDD

def task4(logsRDD, l):
    return logsRDD

def task5(bipartiteGraphRDD):
    return bipartiteGraphRDD

def task6(logsRDD, day1, day2):
    return logsRDD

def task7(nobelRDD):
    return nobelRDD

def task8(flewon, n, m):
    fragment_and_replicate()
    return flewon

File: project7/java/src/main/java/Tasks.java
import java.util.*;
import java.io.Serializable;

import org.apache.spark.api.java.*;
import org.apache.spark.api.java.function.*;
import scala.Tuple2;
import scala.Tuple3;
import scala.Tuple4;
import org.json.*;

public class Tasks {

    public static JavaPairRDD<String, Integer> task1(JavaRDD<String> playRDD) {
        return null;
    }

    public static Iterator<Integer> task2(JSONObject json) {
        return null;
    }

    public static JavaPairRDD<String, Iterable<Integer>> task3(JavaRDD<String> nobelRDD) {
        return null;
    }

    public static JavaRDD<String> task4(JavaRDD<String> logsRDD, List<String> requests) {
        return null;
    }

    public static JavaPairRDD<Long, Long> task5(JavaPairRDD<String, String> amazonBipartiteRDD) {
        return null;
    }

    public static JavaPairRDD<String, Tuple2<Iterable<String>, Iterable<String>>> task6(JavaRDD<String> logsRDD, String host1, String host2) {
        return null;
    }

    public static JavaPairRDD<String, Long> task7(JavaRDD<String> nobelRDD) {
        return null;
    }

    public static JavaRDD<Tuple2<String, Long>> task8(JavaRDD<FlewonTuple> flewonRDD, int n, int m, Function2<Tuple2<String, Double>, Tuple2<String, Double>, Boolean> joinCondition) {
        return null;
    }

}
